client-cli/src/main/java/org/eclipse/dataspaceconnector/registration/cli/RegistrationServiceCli.java

```diff
@@ -63,7 +78,21 @@ private void init() {
         } catch (IOException e) {
             throw new RuntimeException("Error reading file " + privateKeyFile, e);
         }
-        var apiClient = ClientUtils.createApiClient(service, clientDid, privateKeyData);
-        this.registryApiClient = new RegistryApi(apiClient);
+
+        // temporary to preserve the backwards compatibility
+        if (dataspaceDid.isEmpty()) {
+            var apiClient = ClientUtils.createApiClient(service, clientDid, privateKeyData);
+            this.registryApiClient = new RegistryApi(apiClient);
+            return;
+        }
+        var didWebResolver = new WebDidResolver(new OkHttpClient(), useHttpsScheme(), MAPPER, new ConsoleMonitor());
```

Comment:
may we can build http client with timeouts as did in EDC core extension. e.g. https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/39e8da36860a9087eae76fcd3cdae2023b4e30d0/core/base/src/main/java/org/eclipse/dataspaceconnector/core/CoreServicesExtension.java#L208